### ISSUE TITLE
GSS_data/Trade/ONS_ABS #109 failed

### ISSUE THREAD

**Issue by ajtucker** [ONS-OpenData/ONS_ABS#9]
Build 'GSS_data/Trade/ONS_ABS' is failing!

Last 50 lines of build output:

```
[...truncated 1495 lines...]
1332. duplicate_key. Row: 1999. 2-4,,Proportion of all Business,any,Services,any,2016,any,Percent
1333. required. Row: 2003,2
1334. required. Row: 2004,2
1335. duplicate_key. Row: 2004. 4-10,,Count,any,Services,any,2016,any,Businesses
1336. required. Row: 2007,2
1337. required. Row: 2008,2
1338. duplicate_key. Row: 2008. 4-10,,Proportion of all Business,any,Services,any,2016,any,Percent
1339. required. Row: 2012,2
1340. required. Row: 2013,2
1341. duplicate_key. Row: 2013. 10-20,,Count,any,Services,any,2016,any,Businesses
1342. required. Row: 2016,2
1343. required. Row: 2017,2
1344. duplicate_key. Row: 2017. 10-20,,Proportion of all Business,any,Services,any,2016,any,Percent
1345. required. Row: 2021,2
1346. required. Row: 2022,2
1347. duplicate_key. Row: 2022. 20-above,,Count,any,Services,any,2016,any,Businesses
1348. required. Row: 2025,2
1349. required. Row: 2026,2
1350. duplicate_key. Row: 2026. 20-above,,Proportion of all Business,any,Services,any,2016,any,Percent
[Pipeline] }
[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // script
[Pipeline] }
$ docker stop --time=1 faf27afaf313c47dc46a70b762d487bd7862a2471b8b4b6768c169d49cffc6b1
$ docker rm -f faf27afaf313c47dc46a70b762d487bd7862a2471b8b4b6768c169d49cffc6b1
[Pipeline] // withDockerContainer
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Upload Tidy Data)
Stage "Upload Tidy Data" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Test draft dataset)
Stage "Test draft dataset" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Publish)
Stage "Publish" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Declarative: Post Actions)
[Pipeline] script
[Pipeline] {
[Pipeline] step

```

Changes since last successful build:
No changes

[View full output](https://ci.floop.org.uk/job/GSS_data/job/Trade/job/ONS_ABS/109/)

**Comment by ajtucker** [comment]
Build 'GSS_data/Trade/ONS_ABS' is failing!

Last 50 lines of build output:

```
[...truncated 1489 lines...]
1332. duplicate_key. Row: 1999. 2-4,,Proportion of all Business,any,Services,any,2016,any,Percent
1333. required. Row: 2003,2
1334. required. Row: 2004,2
1335. duplicate_key. Row: 2004. 4-10,,Count,any,Services,any,2016,any,Businesses
1336. required. Row: 2007,2
1337. required. Row: 2008,2
1338. duplicate_key. Row: 2008. 4-10,,Proportion of all Business,any,Services,any,2016,any,Percent
1339. required. Row: 2012,2
1340. required. Row: 2013,2
1341. duplicate_key. Row: 2013. 10-20,,Count,any,Services,any,2016,any,Businesses
1342. required. Row: 2016,2
1343. required. Row: 2017,2
1344. duplicate_key. Row: 2017. 10-20,,Proportion of all Business,any,Services,any,2016,any,Percent
1345. required. Row: 2021,2
1346. required. Row: 2022,2
1347. duplicate_key. Row: 2022. 20-above,,Count,any,Services,any,2016,any,Businesses
1348. required. Row: 2025,2
1349. required. Row: 2026,2
1350. duplicate_key. Row: 2026. 20-above,,Proportion of all Business,any,Services,any,2016,any,Percent
[Pipeline] }
[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // script
[Pipeline] }
$ docker stop --time=1 17844b426f88006217c04516c4bf10680c29d8f80943560b05c86051408d6abd
$ docker rm -f 17844b426f88006217c04516c4bf10680c29d8f80943560b05c86051408d6abd
[Pipeline] // withDockerContainer
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Upload Tidy Data)
Stage "Upload Tidy Data" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Test draft dataset)
Stage "Test draft dataset" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Publish)
Stage "Publish" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Declarative: Post Actions)
[Pipeline] script
[Pipeline] {
[Pipeline] step

```

Changes since last successful build:
No changes
- [Alex Tucker] 0f97c400a2ecb1baf0ecdee40116b7ac771784ef - Add test to check for multiple values for a measure.

- [Alex Tucker] a1020510aed3998ecf7626318f42908aa8535a93 - Track the test results.

- [Alex Tucker] 308ec4fc70e7c503088b45694ded2fbcf20ac1cf - Put SPARQL tests in tests/sparql.

- [Alex Tucker] 9164ed5917356fbffaee1c6b4d27a7e0791357fb - Run tests over union with draftset and live as DSD is outside draftset. Restrict test to just this dataset.

- [Alex Tucker] a92678becbe0694f227fffb9c38ee8377e9d5be7 - Refactor to range over each table in the spreadsheet independantly.

- [Alex Tucker] 435703665c446016d2edd24dda22cbdfebe16fdf - Add 'units' and rename columns.

- [Alex Tucker] 5e593a12dab20e633ea3dc54672fa50e9da001de - Rename 'Trade' column to 'ONS ABS Trade'.

- [Alex Tucker] e2e99c17633fbefadccfd727b1856e3141ba1f75 - Observations are now observations.csv.

- [Alex Tucker] 37d6e9ce8675451157a4e0171410a28ee625726e - Units -> Unit.

- [Alex Tucker] d66420326b62528916d88c27ad1aa2936ba1e958 - Measure type 'Percent' -> 'Proportion of all Business'

- [Alex Tucker] 600cbc4cf344f0bb0ce61ffdc15b420dccbd2b24 - Example BDD acceptance test.

- [Alex Tucker] 32384e8cd628e18836fb74d74a54caa825f3a2db - Run acceptance tests.

- [Alex Tucker] 3f8eb596e31578a7208db8b09424e4ef5dbee761 - Use patched behave to output Cucumber reports that Jenkins accepts.

- [Robin Gower] 1a4227d284ce1efd8e2e66f2c8a07cf89b71c1b1 - Add comment including keyword trade

- [Alex Tucker] 802fc9066aba51fb935e3b9531454d350657a661 - Don't stop on failed acceptance test. Use jupyter-nbconvert --execute to run scripts.

- [Robin Gower] 17bd3559d5ab54fa94c09bdd67404d1e817656ef - Use jobDraft.publish()

- [Robin Gower] e8000812b4bfc9c2a2e52490479dc624ad36f926 - Revert label to check coupling

- [Robin Gower] 9efce8eb333b4634a9ec3178754be67362397506 - Add trade theme and family

- [Robin Gower] 34ce1ee4fe56a543c43af32b90c64b52c99cadc3 - Add dcat:theme

- [julija] 86d1ef0c75eeb9e9a66a83467a4f3cb05f19bb8b - Fix syntax error

- [Alex Tucker] e107cf6b3f9f91e93cbf375ca46b2f491050f729 - Use gssutils for scraping metadata and fetching the latest version.

- [Alex Tucker] ce137a1dfe557dcc41a2e0b58b8978ca21adde83 - Remove BDD acceptance tests. Update for newer PMD-Jenkins pipeline. Use csvlint schema test. Auto update Trello.

- [Alex Tucker] 0690bea0eabab2576536fd81c9186c50cfa176a5 - Remove old test harness.

- [Alex Tucker] 5a2cdf36a523b3cee024ef8df081ba6628f4dec8 - Remove old BDD tests.

- [Alex Tucker] 143cadd02e03fe7e9d0fd9e5500595197c014c5b - Remove old metadata creation script.

- [Alex Tucker] a4dd82e7ba39fe7e1fe749e9ab1f98e6750f2634 - Check notation rather than label as codelist doesn't match 'slugged' labels.

- [Vamshi] eeac7539d35f61e30017e01e9d802405c2c91184 - Schema to use notation

- [Vamshi] 7d66dd59db1b9e0f6f56acec5cb35a6033abf98c - Closes: #1

- [Vamshi] 503a5d4340b0b4a140d5fe7f6b389298e9312b75 - Closes: #2,3

- [Vamshi] c148510b9509a55d11b8aa7e4a1739ba52ca7812 - Closes: #4

- [Vamshi] e1f2e9e56d682dc06419d9c962d5fc9cb0eb068f - Closes: #5

- [noreply] 08e9b68431b5d732f8798289d45aba1358695c7b - Update main.py



[View full output](https://ci.floop.org.uk/job/GSS_data/job/Trade/job/ONS_ABS/110/)

**Comment by ajtucker** [comment]
Build 'GSS_data/Trade/ONS_ABS' is failing!

Last 50 lines of build output:

```
[...truncated 1489 lines...]
1332. duplicate_key. Row: 1999. 2-4,,Proportion of all Business,any,Services,any,2016,any,Percent
1333. required. Row: 2003,2
1334. required. Row: 2004,2
1335. duplicate_key. Row: 2004. 4-10,,Count,any,Services,any,2016,any,Businesses
1336. required. Row: 2007,2
1337. required. Row: 2008,2
1338. duplicate_key. Row: 2008. 4-10,,Proportion of all Business,any,Services,any,2016,any,Percent
1339. required. Row: 2012,2
1340. required. Row: 2013,2
1341. duplicate_key. Row: 2013. 10-20,,Count,any,Services,any,2016,any,Businesses
1342. required. Row: 2016,2
1343. required. Row: 2017,2
1344. duplicate_key. Row: 2017. 10-20,,Proportion of all Business,any,Services,any,2016,any,Percent
1345. required. Row: 2021,2
1346. required. Row: 2022,2
1347. duplicate_key. Row: 2022. 20-above,,Count,any,Services,any,2016,any,Businesses
1348. required. Row: 2025,2
1349. required. Row: 2026,2
1350. duplicate_key. Row: 2026. 20-above,,Proportion of all Business,any,Services,any,2016,any,Percent
[Pipeline] }
[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // script
[Pipeline] }
$ docker stop --time=1 b0bf5da3641b26f8cf987890987aa1ce8950f024fd5882a5f9e541741889091d
$ docker rm -f b0bf5da3641b26f8cf987890987aa1ce8950f024fd5882a5f9e541741889091d
[Pipeline] // withDockerContainer
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Upload Tidy Data)
Stage "Upload Tidy Data" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Test draft dataset)
Stage "Test draft dataset" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Publish)
Stage "Publish" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Declarative: Post Actions)
[Pipeline] script
[Pipeline] {
[Pipeline] step

```

Changes since last successful build:
No changes
- [Alex Tucker] 0f97c400a2ecb1baf0ecdee40116b7ac771784ef - Add test to check for multiple values for a measure.

- [Alex Tucker] a1020510aed3998ecf7626318f42908aa8535a93 - Track the test results.

- [Alex Tucker] 308ec4fc70e7c503088b45694ded2fbcf20ac1cf - Put SPARQL tests in tests/sparql.

- [Alex Tucker] 9164ed5917356fbffaee1c6b4d27a7e0791357fb - Run tests over union with draftset and live as DSD is outside draftset. Restrict test to just this dataset.

- [Alex Tucker] a92678becbe0694f227fffb9c38ee8377e9d5be7 - Refactor to range over each table in the spreadsheet independantly.

- [Alex Tucker] 435703665c446016d2edd24dda22cbdfebe16fdf - Add 'units' and rename columns.

- [Alex Tucker] 5e593a12dab20e633ea3dc54672fa50e9da001de - Rename 'Trade' column to 'ONS ABS Trade'.

- [Alex Tucker] e2e99c17633fbefadccfd727b1856e3141ba1f75 - Observations are now observations.csv.

- [Alex Tucker] 37d6e9ce8675451157a4e0171410a28ee625726e - Units -> Unit.

- [Alex Tucker] d66420326b62528916d88c27ad1aa2936ba1e958 - Measure type 'Percent' -> 'Proportion of all Business'

- [Alex Tucker] 600cbc4cf344f0bb0ce61ffdc15b420dccbd2b24 - Example BDD acceptance test.

- [Alex Tucker] 32384e8cd628e18836fb74d74a54caa825f3a2db - Run acceptance tests.

- [Alex Tucker] 3f8eb596e31578a7208db8b09424e4ef5dbee761 - Use patched behave to output Cucumber reports that Jenkins accepts.

- [Robin Gower] 1a4227d284ce1efd8e2e66f2c8a07cf89b71c1b1 - Add comment including keyword trade

- [Alex Tucker] 802fc9066aba51fb935e3b9531454d350657a661 - Don't stop on failed acceptance test. Use jupyter-nbconvert --execute to run scripts.

- [Robin Gower] 17bd3559d5ab54fa94c09bdd67404d1e817656ef - Use jobDraft.publish()

- [Robin Gower] e8000812b4bfc9c2a2e52490479dc624ad36f926 - Revert label to check coupling

- [Robin Gower] 9efce8eb333b4634a9ec3178754be67362397506 - Add trade theme and family

- [Robin Gower] 34ce1ee4fe56a543c43af32b90c64b52c99cadc3 - Add dcat:theme

- [julija] 86d1ef0c75eeb9e9a66a83467a4f3cb05f19bb8b - Fix syntax error

- [Alex Tucker] e107cf6b3f9f91e93cbf375ca46b2f491050f729 - Use gssutils for scraping metadata and fetching the latest version.

- [Alex Tucker] ce137a1dfe557dcc41a2e0b58b8978ca21adde83 - Remove BDD acceptance tests. Update for newer PMD-Jenkins pipeline. Use csvlint schema test. Auto update Trello.

- [Alex Tucker] 0690bea0eabab2576536fd81c9186c50cfa176a5 - Remove old test harness.

- [Alex Tucker] 5a2cdf36a523b3cee024ef8df081ba6628f4dec8 - Remove old BDD tests.

- [Alex Tucker] 143cadd02e03fe7e9d0fd9e5500595197c014c5b - Remove old metadata creation script.

- [Alex Tucker] a4dd82e7ba39fe7e1fe749e9ab1f98e6750f2634 - Check notation rather than label as codelist doesn't match 'slugged' labels.

- [Vamshi] eeac7539d35f61e30017e01e9d802405c2c91184 - Schema to use notation

- [Vamshi] 7d66dd59db1b9e0f6f56acec5cb35a6033abf98c - Closes: #1

- [Vamshi] 503a5d4340b0b4a140d5fe7f6b389298e9312b75 - Closes: #2,3

- [Vamshi] c148510b9509a55d11b8aa7e4a1739ba52ca7812 - Closes: #4

- [Vamshi] e1f2e9e56d682dc06419d9c962d5fc9cb0eb068f - Closes: #5

- [noreply] 08e9b68431b5d732f8798289d45aba1358695c7b - Update main.py

No changes
No changes


[View full output](https://ci.floop.org.uk/job/GSS_data/job/Trade/job/ONS_ABS/112/)

**Comment by ajtucker** [comment]
Build 'GSS_data/Trade/ONS_ABS' is failing!

Last 50 lines of build output:

```
[...truncated 1489 lines...]
1332. duplicate_key. Row: 1999. 2-4,,Proportion of all Business,any,Services,any,2016,any,Percent
1333. required. Row: 2003,2
1334. required. Row: 2004,2
1335. duplicate_key. Row: 2004. 4-10,,Count,any,Services,any,2016,any,Businesses
1336. required. Row: 2007,2
1337. required. Row: 2008,2
1338. duplicate_key. Row: 2008. 4-10,,Proportion of all Business,any,Services,any,2016,any,Percent
1339. required. Row: 2012,2
1340. required. Row: 2013,2
1341. duplicate_key. Row: 2013. 10-20,,Count,any,Services,any,2016,any,Businesses
1342. required. Row: 2016,2
1343. required. Row: 2017,2
1344. duplicate_key. Row: 2017. 10-20,,Proportion of all Business,any,Services,any,2016,any,Percent
1345. required. Row: 2021,2
1346. required. Row: 2022,2
1347. duplicate_key. Row: 2022. 20-above,,Count,any,Services,any,2016,any,Businesses
1348. required. Row: 2025,2
1349. required. Row: 2026,2
1350. duplicate_key. Row: 2026. 20-above,,Proportion of all Business,any,Services,any,2016,any,Percent
[Pipeline] }
[Pipeline] // ansiColor
[Pipeline] }
[Pipeline] // script
[Pipeline] }
$ docker stop --time=1 9ed557fa9f9a228f36968caa01f21fe84bdff285d139ec6e5395e29542208a38
$ docker rm -f 9ed557fa9f9a228f36968caa01f21fe84bdff285d139ec6e5395e29542208a38
[Pipeline] // withDockerContainer
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Upload Tidy Data)
Stage "Upload Tidy Data" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Test draft dataset)
Stage "Test draft dataset" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Publish)
Stage "Publish" skipped due to earlier failure(s)
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Declarative: Post Actions)
[Pipeline] script
[Pipeline] {
[Pipeline] step

```

Changes since last successful build:
No changes
- [Alex Tucker] 0f97c400a2ecb1baf0ecdee40116b7ac771784ef - Add test to check for multiple values for a measure.

- [Alex Tucker] a1020510aed3998ecf7626318f42908aa8535a93 - Track the test results.

- [Alex Tucker] 308ec4fc70e7c503088b45694ded2fbcf20ac1cf - Put SPARQL tests in tests/sparql.

- [Alex Tucker] 9164ed5917356fbffaee1c6b4d27a7e0791357fb - Run tests over union with draftset and live as DSD is outside draftset. Restrict test to just this dataset.

- [Alex Tucker] a92678becbe0694f227fffb9c38ee8377e9d5be7 - Refactor to range over each table in the spreadsheet independantly.

- [Alex Tucker] 435703665c446016d2edd24dda22cbdfebe16fdf - Add 'units' and rename columns.

- [Alex Tucker] 5e593a12dab20e633ea3dc54672fa50e9da001de - Rename 'Trade' column to 'ONS ABS Trade'.

- [Alex Tucker] e2e99c17633fbefadccfd727b1856e3141ba1f75 - Observations are now observations.csv.

- [Alex Tucker] 37d6e9ce8675451157a4e0171410a28ee625726e - Units -> Unit.

- [Alex Tucker] d66420326b62528916d88c27ad1aa2936ba1e958 - Measure type 'Percent' -> 'Proportion of all Business'

- [Alex Tucker] 600cbc4cf344f0bb0ce61ffdc15b420dccbd2b24 - Example BDD acceptance test.

- [Alex Tucker] 32384e8cd628e18836fb74d74a54caa825f3a2db - Run acceptance tests.

- [Alex Tucker] 3f8eb596e31578a7208db8b09424e4ef5dbee761 - Use patched behave to output Cucumber reports that Jenkins accepts.

- [Robin Gower] 1a4227d284ce1efd8e2e66f2c8a07cf89b71c1b1 - Add comment including keyword trade

- [Alex Tucker] 802fc9066aba51fb935e3b9531454d350657a661 - Don't stop on failed acceptance test. Use jupyter-nbconvert --execute to run scripts.

- [Robin Gower] 17bd3559d5ab54fa94c09bdd67404d1e817656ef - Use jobDraft.publish()

- [Robin Gower] e8000812b4bfc9c2a2e52490479dc624ad36f926 - Revert label to check coupling

- [Robin Gower] 9efce8eb333b4634a9ec3178754be67362397506 - Add trade theme and family

- [Robin Gower] 34ce1ee4fe56a543c43af32b90c64b52c99cadc3 - Add dcat:theme

- [julija] 86d1ef0c75eeb9e9a66a83467a4f3cb05f19bb8b - Fix syntax error

- [Alex Tucker] e107cf6b3f9f91e93cbf375ca46b2f491050f729 - Use gssutils for scraping metadata and fetching the latest version.

- [Alex Tucker] ce137a1dfe557dcc41a2e0b58b8978ca21adde83 - Remove BDD acceptance tests. Update for newer PMD-Jenkins pipeline. Use csvlint schema test. Auto update Trello.

- [Alex Tucker] 0690bea0eabab2576536fd81c9186c50cfa176a5 - Remove old test harness.

- [Alex Tucker] 5a2cdf36a523b3cee024ef8df081ba6628f4dec8 - Remove old BDD tests.

- [Alex Tucker] 143cadd02e03fe7e9d0fd9e5500595197c014c5b - Remove old metadata creation script.

- [Alex Tucker] a4dd82e7ba39fe7e1fe749e9ab1f98e6750f2634 - Check notation rather than label as codelist doesn't match 'slugged' labels.

- [Vamshi] eeac7539d35f61e30017e01e9d802405c2c91184 - Schema to use notation

- [Vamshi] 7d66dd59db1b9e0f6f56acec5cb35a6033abf98c - Closes: #1

- [Vamshi] 503a5d4340b0b4a140d5fe7f6b389298e9312b75 - Closes: #2,3

- [Vamshi] c148510b9509a55d11b8aa7e4a1739ba52ca7812 - Closes: #4

- [Vamshi] e1f2e9e56d682dc06419d9c962d5fc9cb0eb068f - Closes: #5

- [noreply] 08e9b68431b5d732f8798289d45aba1358695c7b - Update main.py

No changes
No changes
No changes


[View full output](https://ci.floop.org.uk/job/GSS_data/job/Trade/job/ONS_ABS/113/)

**Comment by ajtucker** [comment]
Build was fixed!
